File: HackerRank/Regex/Easy/E0030.py
# Problem Statement: https://www.hackerrank.com/challenges/detecting-valid-latitude-and-longitude/problem

import re

for _ in range(int(input())):
	print('Valid' if re.match(r'\([+-]?([0-8]\d?(\.\d+)?|90(\.0+)?),\s[+-]?((\d{1,2}|1[0-7]\d)(\.\d+)?|180(\.0+)?)\)$', input()) else 'Invalid')

File: HackerRank/Algorithms/Easy/E0030.cpp
/*
Problem Statement: https://www.hackerrank.com/challenges/kaprekar-numbers/problem
*/

#include <iostream>
#include <vector>
#include <cmath>
#include <string>
#include <algorithm>

using namespace std;

bool checkIfKaprekarNumber(int n){
	int l,r,d;
	long nSquare;
	nSquare = pow(n,2);
	d = to_string(n).length();
	string s = to_string(nSquare);
	r = stoi(s.substr(s.length()-d,d));
	s = s.substr(0,s.length()-d);
	if(s.empty())
		l = 0;
	else
		l = stoi(s);
	if(l+r == n)
		return true;
	return false;
}

void kaprekarNumbers(int p, int q) {
	vector<int> values;
	for(int i=p ; i<=q ; i++)
		if(checkIfKaprekarNumber(i))
			values.push_back(i);
	for(int value:values)
		cout<<value<<" ";
	if(!values.size())
		cout<<"INVALID RANGE"<<endl;
}

int main()
{
	int p,q;
	cin>>p>>q;
	kaprekarNumbers(p,q);
	return 0;
}
